libs/portals/shared-modules/delegations/src/components/access/AccessCard.tsx

```diff
@@ -266,57 +254,61 @@ export const AccessCard = ({
               </Inline>
             </Box>
           )}
-          {showActions && (
+          {hasActions && (
             <Box
               display="flex"
               alignItems="center"
               justifyContent={['spaceBetween', 'flexEnd']}
               marginTop={[2, 0]}
               marginLeft={[0, 3]}
             >
-              {canDelete && (
+              {onDelete && (
                 <Button
                   variant="text"
                   icon="trash"
                   iconType="outline"
                   size="small"
                   colorScheme="destructive"
-                  onClick={() => onDelete(delegation)}
+                  onClick={() => onDelete?.(delegation)}
```

Comment:
Should be unnecessary to optional chain the function call as you have checked that `onDelete` is defined. 
```suggestion
                  onClick={() => onDelete(delegation)}
```